File: src/javascript/user/addressbook/update-address-book.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to update an address book.');
}

const client = new FaxCoreClient();
const result = await client.post('/api/addressbook/update', {
  bookID: Number(process.env.FAXCORE_ADDRESS_BOOK_ID || 0),
  addressBookName: process.env.FAXCORE_ADDRESS_BOOK_NAME || 'Sample Address Book'
});

printResult('Update address book', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/AddressBook/UpdateAddressBook.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.AddressBook
{
    internal static class UpdateAddressBook
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            RequireDestructiveConfirmation();
            return client.PostAsync("/api/addressbook/update", new
            {
                bookID = config.AddressBookID,
                addressBookName = config.AddressBookName
            });
        }

        private static void RequireDestructiveConfirmation()
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to update an address book.");
            }
        }
    }
}
